ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```
@@ -627,18 +608,27 @@ public CompactorMR getMrCompactor() {
     return new CompactorMR();
   }
 
-  private void markFailed(CompactionInfo ci, String errorMessage) {
-    if (ci != null && StringUtils.isNotBlank(errorMessage)) {
+  private void markErrorous(CompactionInfo ci, String errorMessage, boolean failed) {
+    if (ci == null) {
+      LOG.warn("Could not set compaction info state because it is null!");
+      return;
```

maybe throw an exception here?